dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -708,4 +764,68 @@ private Integer uninstall() {
         }
         return OK;
     }
-}
+
+    private static void populateShortNameFromManifest(File esa, Map<String, String> shortNameMap) throws IOException {
+        String esaLocation = esa.getAbsolutePath();
+        ZipFile zip = null;
+        try {
+            zip = new ZipFile(esaLocation);
+            Enumeration<? extends ZipEntry> zipEntries = zip.entries();
+            ZipEntry subsystemEntry = null;
+            while (zipEntries.hasMoreElements()) {
+                ZipEntry nextEntry = zipEntries.nextElement();
+                if ("OSGI-INF/SUBSYSTEM.MF".equalsIgnoreCase(nextEntry.getName())) {
+                    subsystemEntry = nextEntry;
+                    break;
+                }
+            }
+            if (subsystemEntry != null) {
+                Manifest m = ManifestProcessor.parseManifest(zip.getInputStream(subsystemEntry));
+                Attributes manifestAttrs = m.getMainAttributes();
+                String shortNameAttr = manifestAttrs.getValue(SHORTNAME_HEADER_NAME);
+                shortNameMap.put(esa.getAbsolutePath(), shortNameAttr);
+            }
+        } finally {
+            if (zip != null) {
+                zip.close();
+            }
+        }
+    }
+
+    //accept List of Files
+    private File individualESAInstall(Path generatedJson, Map<String, String> shortNameMap) throws IOException, BuildException, RepositoryException, InstallException {
```

Comment:
Rename parameter `generatedJson` since the parameter is not a json but is just the temp dir where the jsons will be stored.  Perhaps use `jsonDirectory`